Keyboard/leet_keyboard.ino

```diff
@@ -117,46 +133,73 @@ void playNotes() {
           Serial.print(F("NoteOff Ch:"));
           Serial.print(midiCh);
           Serial.print(F(" Note:"));
-          Serial.println(note, HEX);
+          Serial.println(pressed[i], HEX);
         }
       }
     }
   }
 }
 
 void checkMidi() {
-  midiEventPacket_t rx;
-  do {
-    rx = MidiUSB.read();
-    if (rx.header != 0) {
-      if (rx.byte1 == (0x90 | midiCh)) {            // tone on
-        strip.SetPixelColor(noteLedMap[(rx.byte2 - 0) % 12], wheelH(map(rx.byte2, 0, 128, 0, 767)));     // replace 0 with 4 if using sunvox MIDI echo...
+  // Forward THRU -> USB
+  while (THRU.read()) {
+    MIDI.send(THRU.getType(),
+              THRU.getData1(),
+              THRU.getData2(),
+              THRU.getChannel());
+
+    switch (THRU.getType()) {
+      case midi::NoteOn:
+        strip.SetPixelColor(noteLedMap[THRU.getData1() % 12], wheelH(map(THRU.getData2(), 0, 128, 0, 767)));     // replace 0 with 4 if using sunvox MIDI echo...
         strip.Show();
-      } else if (rx.byte1 == (0x80 | midiCh)) {     // tone off
-        strip.SetPixelColor(noteLedMap[(rx.byte2 - 0) % 12], RgbColor(0, 0, 0));   // replace 0 with 4 if using sunvox MIDI echo...
+        break;
+      case midi::NoteOff:
+        strip.SetPixelColor(noteLedMap[THRU.getData1() % 12], RgbColor(0, 0, 0));   // replace 0 with 4 if using sunvox MIDI echo...
         strip.Show();
-      }
-      if (debug == true) {
-        Serial.print(F("Received: "));
-        Serial.print(rx.header, HEX);
-        Serial.print("-");
-        Serial.print(rx.byte1, HEX);
-        Serial.print("-");
-        Serial.print(rx.byte2, HEX);
-        Serial.print("-");
-        Serial.println(rx.byte3, HEX);
-      }
+        break;
+    }
+
+    if (debug == true) {
+      Serial.print(F("Received: "));
+      Serial.print(THRU.getType(), HEX);
+      Serial.print("-");
+      Serial.print(THRU.getData1(), HEX);
+      Serial.print("-");
+      Serial.print(THRU.getData2(), HEX);
+      Serial.print("-");
+      Serial.println(THRU.getChannel(), HEX);
     }
-  } while (rx.header != 0);
+  }
+
+  // Update LEDs for MIDI received from USB
+  while (MIDI.read()) {
+    switch (THRU.getType()) {
```

Comment:
This should probably be `MIDI.getType`, otherwise you're reading from the wrong port.